plugins/filetrees/yazi.nix

```diff
@@ -0,0 +1,160 @@
+          cycle_open_buffers = "<tab>";
+          copy_relative_path_to_selected_files = "<c-y>";
+          send_to_quickfix_list = "<c-q>";
+        }
+        ''
+          Customize the keymaps that are active when yazi is open and focused.
+
+          Also:
+            - use e.g. `open_file_in_tab = false` to disable a keymap
+            - you can customize only some of the keymaps if you want
+            - Set to `false` to disable all default keymaps.
+        '';
+
+    set_keymappings_function = defaultNullOpts.mkLuaFn null ''
+      Completely override the keymappings for yazi. This function will be
+      called in the context of the yazi terminal buffer.
+    '';
+
+    hooks = {
+      yazi_opened = defaultNullOpts.mkLuaFn' {
+        pluginDefault.__raw = ''
+          function(preselected_path, yazi_buffer_id, config)
+          end
+        '';
+        description = ''
+          If you want to execute a custom action when yazi has been opened,
+          you can define it here.
+        '';
+      };
+
+      yazi_closed_successfully = defaultNullOpts.mkLuaFn' {
+        pluginDefault.__raw = ''
+          function(chosen_file, config, state)
+          end
+        '';
+        description = "When yazi was successfully closed";
+      };
+
+      yazi_opened_multiple_files = defaultNullOpts.mkLuaFn' {
+        pluginDefault.__raw = ''
+          function(chosen_files)
+            vim.cmd("args" .. table.concat(chosen_files, " "))
+          end
+        '';
+        description = ''
+          When yazi opened multiple files. The default is to send them to the
+          quickfix list, but if you want to change that, you can define it here
+        '';
+      };
+    };
+
+    highlight_groups = defaultNullOpts.mkAttributeSet { hovered_buffer = null; } ''
+      Add highlight groups to different yazi events.
+
+      NOTE: this only works if `use_ya_for_events_reading` is enabled, etc.
+    '';
+
+    floating_window_scaling_factor =
+      defaultNullOpts.mkNum 0.9
+        "The floating window scaling factor. 1 means 100%, 0.9 means 90%, etc.";
+
+    yazi_floating_window_winblend = defaultNullOpts.mkNullableWithRaw' {
+      type = types.ints.between 0 100;
+      pluginDefault = 0;
+      description = "`0` for fully opaque and `100` for fully transparent. See :h winblend";
+    };
+
+    yazi_floating_window_border = defaultNullOpts.mkBorder "rounded" "yazi" ''
+      The type of border to use for the floating window.
+
+      Supports all available border types from `vim.api.keyset.win_config.border`.
+    '';
+  };
+
+  settingsExample = {
+    log_level = "debug";
+    open_for_directories = true;
+    enable_mouse_support = true;
+    floating_window_scaling_factor = 0.5;
+    yazi_floating_window_border = "single";
+    yazi_floating_window_winblend = 50;
+  };
+
+  # TODO: remove when https://github.com/NixOS/nixpkgs/pull/333506 is available
+  extraPlugins = with pkgs.vimPlugins; [ plenary-nvim ];
```

Comment:
I have a feeling #333506 will merge soon as it's already got one approval. Shall we just wait until we update to a nixpkgs rev with the change before merging so we don't have to drop this line in another PR?

Don't really mind either way.

Reply:
Usually a couple days delay before getting to unstable after merge and then whenever we get to flake update. I dont mind keeping track of the nixpkgs PR and when we do another flake.lock update. I usually do these kind of TODO's in my own config with overlays while waiting for fixes from nixos-unstable.